container/src/main/java/com/flipkart/poseidon/core/PoseidonServlet.java

```diff
@@ -148,20 +151,26 @@ protected void doRequest(HttpMethod method, HttpServletRequest httpRequest, Http
 
     private void setRequestContext(HttpServletRequest httpServletRequest) {
         RequestContext.set(METHOD, httpServletRequest.getMethod());
-        RequestContext.set(REQUEST_ID, getRequestId(httpServletRequest));
-        RequestContext.set(IS_PERF_TEST, isPerfTest(httpServletRequest));
```

Comment:
Do we need to remove request id and is perf test from existing RequestContext. May be we should populate both ServiceContext and RequestContext with the configured headers.